SalesforceCommerceCloud/pwa-kit
# Extract default messages script

# Description
This PR created a build-translation scripts that helps to extract messages from retail-template and extended project and output into translations folder
The script will ignore extracting messages from any files that are overridden. 
Move translations folder from inside app to root directory



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- pull the request
- npm ci
- rn -rf generated-project
- run `node packages/pwa-kit-create-app/scripts/create-mobify-app-dev.js --outputDir generated-project` to generate a project
- change the extract-default-translations as below
 ```
    "extract-default-translations": "node ./node_modules/retail-react-app/scripts/extract-default-messages.js",
```
- npm run build-translations
- confirm that all translations are extracted and compiled under generated-projected/translations/en-US.json

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
